vegafusion-common/src/data/table.rs

```diff
@@ -271,6 +272,13 @@ impl VegaFusionTable {
         }
     }
 
+    #[cfg(feature = "pyarrow")]
+    pub fn from_arrow_c_stream(table: &Bound<PyAny>) -> PyResult<Self> {
```

Comment:
Is this a public API or is this an internal API called by your Python code? If it's a public API, it might be preferred to overload the existing `from_pyarrow` method. 

You can check for both the pycapsule interface and fall back to pyarrow in a single method. That's a recommended approach [here](https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#backwards-compatibility-with-pyarrow). 

You can do this by first trying to extract a `PyTable`; this will work for any recent pyarrow table object (or pyarrow record batch reader). And then you can fallback to your existing `from_pyarrow` code.

```rs
    pub fn from_arrow(table: &Bound<PyAny>) -> PyResult<Self> {
        if let Ok(table) = data.extract::<PyTable>() {
            let (batches, schema) = table.into_inner();
            Ok(VegaFusionTable::try_new(schema, batches)?)
        } else {
            Self::from_pyarrow(table)
        }
    }
```